service/datastore/internal/src/main/java/org/eclipse/kapua/service/datastore/internal/converter/ModelContextImpl.java

```diff
@@ -127,7 +127,11 @@ private DatastoreMessage unmarshalDatastoreMessage(Map<String, Object> messageMa
         
         KapuaId scopeId = new KapuaEid(new BigInteger((String) messageMap.get(MessageSchema.MESSAGE_SCOPE_ID)));
         message.setScopeId(scopeId);
-        KapuaId deviceId = new KapuaEid(new BigInteger((String) messageMap.get(MessageSchema.MESSAGE_DEVICE_ID)));
+        KapuaId deviceId = null;
+        String deviceIdStr = (String) messageMap.get(MessageSchema.MESSAGE_DEVICE_ID);
+        if (deviceIdStr != null) {
+            deviceId = new KapuaEid(new BigInteger(deviceIdStr));
```

Comment:
Sorry, I overlooked that it was a string. Too bad. because in most cases this will be in the cache I guess. 

We should consider making KapuaEid#valueOf(String) method. But this is outside of the scope of this PR.